contrib/pkg/awsprivatelink/enable.go

```diff
@@ -0,0 +1,219 @@
+	return cmd
+}
+
+func (o *enableOptions) Complete(cmd *cobra.Command, args []string) error {
+	// Get controller-runtime dynamic client
+	if err := configv1.Install(scheme.Scheme); err != nil {
+		log.WithError(err).Fatal("Failed to add Openshift configv1 types to the default scheme")
+	}
+	dynamicClient, err := hiveutils.GetClient()
+	if err != nil {
+		log.WithError(err).Fatal("Failed to create controller-runtime client")
+	}
+	o.dynamicClient = dynamicClient
+
+	// Get current user's home directory
+	o.homeDir = "."
+	u, err := user.Current()
+	if err != nil {
+		log.WithError(err).Fatal("Failed to get the current user")
+	}
+	o.homeDir = u.HomeDir
+
+	// Get AWS clients
+	o.region, o.infraId, err = o.getRegionAndInfraId()
+	if err != nil {
+		log.WithError(err).Fatal("Failed to get region and infraID from Infrastructure/cluster")
+	}
+	log.Debugf("Found region = %v, infraId = %v", o.region, o.infraId)
+	awsClients, err := awsclient.NewClientFromSecret(nil, o.region)
+	if err != nil {
+		log.WithError(err).Fatal("Failed to create AWS clients")
+	}
+	o.awsClients = awsClients
+
+	return nil
+}
+
+func (o *enableOptions) Validate(cmd *cobra.Command, args []string) error {
+	switch o.dnsRecordType {
+	case "Alias", "ARecord":
+	default:
+		log.Fatal(`--dns-record-type must be one of "Alias", "ARecord"`)
+	}
+
+	return nil
+}
+
+func (o *enableOptions) Run(cmd *cobra.Command, args []string) error {
+	// Get HiveConfig
+	hiveConfig := &hivev1.HiveConfig{}
+	if err := o.dynamicClient.Get(context.Background(), types.NamespacedName{Name: "hive"}, hiveConfig); err != nil {
+		log.WithError(err).Fatal("Failed to get HiveConfig/hive")
+	}
+	if hiveConfig.Spec.AWSPrivateLink != nil {
+		log.Fatal("AWS Private Link is already enabled")
+	}
+
+	// Get active cluster's VPC, filtering by infra-id
+	describeVPCsOutput, err := o.awsClients.DescribeVpcs(&ec2.DescribeVpcsInput{
+		Filters: []*ec2.Filter{
+			{
+				Name:   aws.String("tag:Name"),
+				Values: []*string{aws.String(o.infraId + "-vpc")},
+			},
+		},
+	})
+	if err != nil {
+		log.WithError(err).Fatal("Failed to get VPC of the active cluster")
+	}
+	if len(describeVPCsOutput.Vpcs) == 0 {
+		log.Fatal("VPC of the active cluster not found")
+	}
+	vpcID := *describeVPCsOutput.Vpcs[0].VpcId
+	log.Debugf("Found VPC ID = %v for the active cluster", vpcID)
+
+	// Create Hub account credentials Secret
+	hiveNS := hiveConfig.Spec.TargetNamespace
+	if hiveNS == "" {
+		hiveNS = constants.DefaultHiveNamespace
+	}
+	hubCredentialsSecret, err := o.generateAWSCredentialsSecret(hiveNS)
+	if err != nil {
+		log.WithError(err).Fatal("Failed to generate Secret with AWS credentials")
+	}
+	if err = o.dynamicClient.Create(context.Background(), hubCredentialsSecret); err != nil {
+		log.WithError(err).Fatal("Failed to create Secret with hub account credentials")
+	}
+	log.Info("Secret with hub account credentials created")
```

Comment:
nit: Let's have this print the ns/name of the secret.

Reply:
Okay, fixed.